### PR TITLE
Refactoring / clean-ups

### PR DESCRIPTION
**Pull request by max-leuthaeuser**
Mostly from the Scala 3 compiler (warnings) and Intellij linting.

### CODE REVIEW

File: semanticcpg/src/test/scala/io/shiftleft/semanticcpg/language/StepsTest.scala [hunk 1]
```diff
@@ -9,11 +9,11 @@ import org.json4s._
 import org.json4s.native.JsonMethods.parse
 import org.scalatest.matchers.should.Matchers
 import org.scalatest.wordspec.AnyWordSpec
-import overflowdb.traversal.{Traversal, jIteratortoTraversal, toElementTraversal, toNodeTraversal}
+import overflowdb.traversal.Traversal
 
 class StepsTest extends AnyWordSpec with Matchers {
 
-  val cpg = MockCpg()
+  private val cpg: Cpg = MockCpg()
```
Comment (mpollmeier) [inline semanticcpg/src/test/scala/io/shiftleft/semanticcpg/language/StepsTest.scala:16]:
```suggestion
  private val cpg = MockCpg()
```
scalac probably asked for a type anntation here, but that's mostly for public members...

Comment (max-leuthaeuser) [reply semanticcpg/src/test/scala/io/shiftleft/semanticcpg/language/StepsTest.scala:16]: And it results in faster compilation times.

File: joern-cli/frontends/x2cpg/src/main/scala/io/joern/x2cpg/passes/controlflow/cfgcreation/CfgCreator.scala [hunk 2]
```diff
@@ -290,8 +290,8 @@ class CfgCreator(entryNode: Method, diffGraph: DiffGraphBuilder) {
     */
   protected def cfgForConditionalExpression(call: Call): Cfg = {
     val conditionCfg = cfgFor(call.argument(1))
-    val trueCfg      = call.argumentOption(2).map(cfgFor).getOrElse(Cfg.empty)
-    val falseCfg     = call.argumentOption(3).map(cfgFor).getOrElse(Cfg.empty)
+    val trueCfg      = call.argumentOption(2).fold(Cfg.empty)(cfgFor)
```
[the remark this answers is not in the document]
Comment (max-leuthaeuser) [reply joern-cli/frontends/x2cpg/src/main/scala/io/joern/x2cpg/passes/controlflow/cfgcreation/CfgCreator.scala:293]: That's actually debatable, yes. I did the same refactoring at other locations already so I would like to keep it that way for consistency.

File: semanticcpg/src/test/scala/io/shiftleft/semanticcpg/language/StepsTest.scala (hunk 1 above)
Comment (mpollmeier) [reply semanticcpg/src/test/scala/io/shiftleft/semanticcpg/language/StepsTest.scala:16]: true that

**Review by mpollmeier — APPROVED** [review]

**Pull request merged**